Engine/UserInput.cs

```diff
@@ -132,12 +134,49 @@ void IUserInputInterface.RequestUserInput(object dataObject, TimeSpan timeout, b
                     {
                         readerThread = TapThread.Start(() =>
                         {
-                            while (true)
-                                lines.Add(Console.ReadLine());
+                            try
+                            {
+                                var sb = new StringBuilder();
+                                while (true)
+                                {
+                                    
+                                    var chr = Console.ReadKey(true);
+                                    if (chr.KeyChar == 0)
+                                        continue;
+                                    if (chr.Key != ConsoleKey.Enter && chr.Key != ConsoleKey.Backspace)
+                                        Console.Write(IsSecure ? '*' : chr.KeyChar);
+                                    
+                                    if (chr.Key == ConsoleKey.Enter)
+                                    {
+                                        lines.Add(sb.ToString());
+                                        sb.Clear();
+                                        Console.WriteLine();
+                                    }
+
+                                    if (chr.Key == ConsoleKey.Backspace)
+                                    {
+                                        if(sb.Length > 0)
+                                            sb.Remove(sb.Length - 1, 1);
+                                        
+                                        // delete current char and move back.
+                                        Console.Write("\b \b");
```

Comment:
It is possible to delete characters from the line which are not part of the current input. This doesn't affect the input, but it is a bit weird in terms of usability.